src/Avalonia.Base/Input/GamepadEventArgs.cs

```diff
@@ -0,0 +1,277 @@
+        {
+            return LeftAnalogStick.Equals(state.LeftAnalogStick) &&
+                   RightAnalogStick.Equals(state.RightAnalogStick) &&
+                   EqualityComparer<GamepadButtons>.Default.Equals(_buttons, state._buttons);
+        }
+
+        public override int GetHashCode()
+        {
+#if NETSTANDARD
+            HashCode hc = new();
+            hc.Add(LeftAnalogStick);
+            hc.Add(RightAnalogStick);
+            hc.Add(_buttons);
+            return hc.ToHashCode();
+#else
+            return HashCode.Combine(LeftAnalogStick.GetHashCode(), RightAnalogStick.GetHashCode(), _buttons.GetHashCode());
+#endif
+        }
+
+        public static bool operator ==(GamepadState left, GamepadState right)
+        {
+            return left.Equals(right);
+        }
+
+        public static bool operator !=(GamepadState left, GamepadState right)
+        {
+            return !(left == right);
+        }
+    }
+
+    public enum GamepadButton
+    {
+        /// <summary>
+        /// X (PS5), A (Xbox), B (Switch)
+        /// </summary>
+        Button0,
+        /// <summary>
+        /// Circle (PS5), B (Xbox), A (Switch)
+        /// </summary>
+        Button1,
+        /// <summary>
+        /// Square (PS5), X (Xbox), Y (Switch)
+        /// </summary>
+        Button2,
+        /// <summary>
+        /// Triangle (PS5), Y (Xbox), X (Switch)
+        /// </summary>
+        Button3,
+        /// <summary>
+        /// L1 (PS5), LB (Xbox), L (Switch)
+        /// </summary>
+        Button4,
+        /// <summary>
+        /// R1 (PS5), RB (Xbox), R (Switch)
+        /// </summary>
+        Button5,
+        /// <summary>
+        /// L2 (PS5), LT (Xbox), ZL (Switch)
+        /// </summary>
+        Button6,
+        /// <summary>
+        /// R2 (PS5), RT (Xbox), ZR (Switch)
+        /// </summary>
+        Button7,
+        /// <summary>
+        /// Share (PS5), Back (Xbox), Minus (Switch)
+        /// </summary>
+        Button8,
+        /// <summary>
+        /// Options (PS5), Start (Xbox), Plus (Switch)
+        /// </summary>
+        Button9,
+        /// <summary>
+        /// Left Analog Stick Depressed
+        /// </summary>
+        Button10,
+        /// <summary>
+        /// Right Analog Stick Depressed
+        /// </summary>
+        Button11,
+        /// <summary>
+        /// Dpad Up
+        /// </summary>
+        Button12,
+        /// <summary>
+        /// Dpad Down
+        /// </summary>
+        Button13,
+        /// <summary>
+        /// Dpad Left
+        /// </summary>
+        Button14,
+        /// <summary>
+        /// Dpad Right
+        /// </summary>
+        Button15,
+    }
+
+#pragma warning disable CA1823 // Avoid unused private fields
+#pragma warning disable CA1815 // should implement equals
+    // this struct is a glorified fixed-buffer and these "unused fields" will be used. 
+    public struct GamepadButtons
+    {
+        private ButtonState _0;
```

Comment:
I think, we can use actual fixed-buffer structs on netstandard.

Reply:
Maybe, but we don't have MemoryMarshal.CreateSpan<T>(T, Int32) in .Net Standard 2.0 

So I would still need to venture off the happy-path for spans to keep NS2.0 compatibility. :(